### ISSUE TITLE
재현성 관련 문의

### ISSUE THREAD

**Issue by knowin-kyeong** [LISTatSNU/FastMRI_challenge#420]
안녕하세요. Placeholder 팀의 서주원 학부생입니다. 

저희 팀이 작년에 대회에 참여했을 때부터, iter 56쯤 부터 1e-4 단위로 train loss log가 약간 달라지는 문제를 마주하고 있습니다. 아마 Prompt block 자체적인 비결정적 연산으로 인해 발생하는 문제로 추정하고 있습니다. Baseline seed_fix() 수준의 seed 고정 + external library seed를 모두 고정하였음에도 동일한 문제가 재발하며, 심지어 Data augmentation 등 추가적으로 구현한 부분을 제외하여 기초적인 Prompt block을 사용해도 동일한 문제가 재발하는 상황입니다.

저희 팀은 이러한 현상이 seed fix 누락이 아닌 Promptmr block 자체의 non-deterministric 연산 때문인 것으로 추측하고 있습니다.
(F.interpolate(bilinear) 등과 같은 모델 자체 구현의 비결정적 연산들 때문에 발생하는 문제로 추정하고 있는 상황이며, seed fix, baseline 수준의 cudnn deterministric 강제로는 고쳐지지 않고 있습니다.)

또한, PromptMR 계열을 사용한 작년 참가팀의 수상 후기와 재현 결과 관련 문의를 보니 저희 팀만의 문제가 아닌 것으로 생각하고 있습니다.

1. #354 와 같이 작년 대회의 경우에는 작년 조교분들이 위 현상을 감안하여 평가를 진행한 것으로 이해하고 있습니다.

2. 실제로 비슷한 모델을 사용한 수상자 후기를 참고했더니 다음과 같이 train loss의 1e-4 수준의 loss log difference가 발생할 수 있음이 언급되었습니다.
<img width="1676" height="230" alt="Image" src="https://github.com/user-attachments/assets/153445fd-d8b6-408d-bdd6-a873da69a02c" />

저희 팀이 외부 환경만 사용하다 GTX 1080을 오늘에서야 처음 대여받았기 때문에 남은 시간/하드웨어 제약으로 2주 정도의 schedule을 가지는 promptmr 계열 모델을 끝까지 **여러 번** 돌려서 final ssim score가 소수점 네 자리까지 **완벽하게 일치**함을 실험하기 어려운 상황입니다. (저희가 사용했던 다른 gpu에서는 final ssim이 네 자리까지 일치함을 여러 번 실험하였고, GTX 1080의 경우에도 당연히 크게 달라지지 않을 것이 확실하지만 소수점 네 자리에서 미세한 변동이 생길 수도 있을 것 같습니다.)

따라서 Rule book에 의거한 **추가적인 엄격한 재현성 검증 절차**에 대비하려고 합니다. 위 절차가 어떻게 이루어지는지 여쭙고 싶습니다.

#354 처럼 작년과 동일하게 train loss log가 몇 step부터 달라지는지 언급하고 관련 사항을 언급하는 정도로 실격을 피할 수 있을지,
아니면 추가적으로 대비해야 하는 상황이 있는지 궁금합니다.

긴 글 읽어 주셔서 감사합니다.
Placeholder 팀, 서주원 올림

**Comment by knowin-kyeong** [comment]
+ LLM을 통해 F.interpolate(bilinear) 대신 같은 결과를 내뱉는 deterministric custom 구현으로 대체하면 전날 언급했던 문제 (train loss 불일치) 를 완화할 수 있으니, 저와 비슷한 고민을 하는 팀들의 경우 참고하시면 좋을 것 같습니다..!

**Comment by hwater27** [comment]
@knowin-kyeong 안녕하세요?

질문해주신 사항에 대해서는 운영진 또한 인지하고 있어, 주요 학습 코드 실행에 차질이 생기는 수준(예: 학습 전과정을 반복적으로 실행해서 재현성 실측)의 검증을 요구하지는 않습니다. 제공된 코드의 `val_loss_log.npy`에 더해 train loss의 terminal output (`logging` 등 이용), 추가로 작년과 같이 학습 초기에 한해서 추가 자료를 준비해주시면 됩니다. 운영진이 직접 서버에 접근할 수 있으니, end-to-end 실행 터미널 자체를 서버에 그대로 남겨두시는 것도 검증을 위한 좋은 방안입니다.

최종 제출에 관해서는 전체 참여자 대상으로 별도 공지가 제공될 예정입니다.
